File: atcoder/abc140B_buffet.nim
# unihernandez22
# https://atcoder.jp/contests/abc140/tasks/abc140_b
# implementation

import strutils, sequtils

let
  n = stdin.readLine.parseInt
  a = stdin.readLine.split.map(parseInt)
  b = stdin.readLine.split.map(parseInt)
  c = stdin.readLine.split.map(parseInt)

var
  last = -2
  ans = 0
  i: int
for x in a:
  i = x-1
  ans += b[i]
  if last == i-1:
    ans += c[i-1]
  last = i

ans.echo

File: atcoder/abc140B_buffet.ml
(* vicfred
 * https://atcoder.jp/contests/abc140/tasks/abc140_b
 * implementation
 * *)

let f () = Scanf.scanf " %d" ( + ) 0
let n = f ()
let a = Array.init n @@ fun _ -> f ()
let b = Array.init n @@ fun _ -> f ()
let c = Array.init (n - 1) @@ fun _ -> f ()
let ans = ref @@ Array.fold_left ( + ) 0 b
let main =
    for i = 0 to (n - 2) do
        if a.(i + 1) = a.(i) + 1 then ans := !ans + c.(a.(i) - 1)
    done;
    Printf.printf "%d\n" !ans

